File: agent/js/examples/minimal.ts
import Agent, {
  type StepResult,
  type StepHandler,
  type TaskInput,
  type StepInput,
} from 'agent-protocol'

const taskHandler = async (
  taskInput: TaskInput | null
): Promise<StepHandler> => {
  console.log(`task: ${taskInput}`)

  const stepHandler = async (
    stepInput: StepInput | null
  ): Promise<StepResult> => {
    console.log(`step: ${stepInput}`)
    return {
      output: stepInput,
    }
  }

  return stepHandler
}

Agent.handleTask(taskHandler).start()

File: agent/python/examples/minimal.py
from agent_protocol import Agent, Task, Step


async def task_handler(task: Task) -> None:
    print(f"task: {task.input}")
    await Agent.db.create_step(task.task_id, task.input)


async def step_handler(step: Step) -> Step:
    print(f"step: {step.input}")
    await Agent.db.create_step(step.task_id, f"Next step from step {step.name}")
    step.output = step.input
    return step


Agent.setup_agent(task_handler, step_handler).start()
